### ISSUE TITLE
game won't start no usefull logfile Information

### ISSUE THREAD

**Issue by Legodev** [virtual-programming/dirtshowdown-linux#29]
I've bought the Game 2 days ago, but did not manage to launch it using Linux. The games shows an black screen for 4 Seconds, then the game crashed and crash dialogue does appear, there I

 also submitted my problem and got the response code:  4JHUZTWC1H

I've validated the Gamecache using Steam and also studied the log file but it did not find anything useful inside ... 

To workaround the first Problem (see first Report): "[thread ffffffff][I][122]: setrlimit(RLIMIT_MEMLOCK) failed, (1) Operation not permitted" 
i run: " sudo sh -c "ulimit -l unlimited && exec su $LOGNAME" "

This did help (see second quote) to remove the errors of thread ffffffff.
But the Game still does not successful launch. 
Maybe its  a bug caused by an Multimonitor setup.

During the Debugging tests I run the game 15 Minutes, so there is plenty of time to get an refund if it will not run. Because of this I can run several other tests if you tell me what you want to be tested.

I am aware that the game uses mono, wine or something else, but it is very curious that the game make extensive use of the Windows API and specifics and floods the logfile with these useless informations. 
Do not get me wrong I want the game but without an good log output no Linux user can find Problems, which may be caused by an Incompatibility of the Game with the System... I do not expect an total port of the game, but since you advertise the Linux support, there should be at least an useful Logfile.

Report 1:
[Crash Report Mon 08 Feb 2016 05:54:09 PM CET.txt](https://github.com/virtual-programming/dirtshowdown-linux/files/121950/Crash.Report.Mon.08.Feb.2016.05.54.09.PM.CET.txt)
Report 2:
[Crash Report Mon 08 Feb 2016 06:02:06 PM CET.txt](https://github.com/virtual-programming/dirtshowdown-linux/files/121949/Crash.Report.Mon.08.Feb.2016.06.02.06.PM.CET.txt)

**Comment by vpltd-kgalaj** [comment]
In the logs you attached, there are mentions of the crash dumps ( example: "Crash Dump:
46541821-bc9a-4cdc-502e8cd5-67984f00.dmp (523432 bytes)" ). These are typically uploaded to our server, and the end user is told the ID (string of alphanumeric characters) of the crash dump, so we can find it on the server. That is, a dialog box likely showed up, identifying the crash dump and mentioning to advise it in techincal support emails. Could you advise this information, so I can look at the crash dumps? The log is just additional information to the crash dump.

**Comment by vpltd-kgalaj** [comment]
Ah, sorry. I see you mentioned one of the IDs, and looked at it. It shows that the thread that crashed was in the middle of allocating memory, and crashed in the original Windows binary code (to which we usually don't have the source code available) but otherwise provides no additional information as to what might have been its cause.

The MEMLOCK limit is fine, the game should work fine without enabling the it to use the memory. It's just an optional attempt by the game.

The porting engine didn't support multimonitor setups well at the time when the game was ported, so it might be a bug with it.

If there were any issues about incompatibilities of the porting engine with the system, they would have shown up in the log as well. I've read the log, and the messages it shows are all typical, or non-critical warnings, so nothing there gives me a clue as to the reason for the crash. Personally, I'd check if the situation changes with a single monitor next.

**Comment by Legodev** [comment]
Well then I will test a single monitor setup next. Is there a way to enable the window mode without starting the game? This might help too if it is an allocation problem with a multi monitor desktop.

**Comment by vpltd-kgalaj** [comment]
In newest version of the porting engine, there's a command-line switch --eon_force_display=x, where x is 0,1,... , but I'm not sure if it was already in place when Dirt was last updated. It basically hides all the displays but the forced one from the game.

**Comment by Legodev** [comment]
This time I started a second Xorg Server and disabled all screens excluding one. Its the same behaviour: STL5XF2G1Z

Thanks for your help :)

**Comment by vpltd-kgalaj** [comment]
Hmm. Dirt is using uncommonly many threads, among games. And I see that the second crash you provided is having the 64th thread crash. It's an unusual coincidence, and an unusually round number... possibly there's some limitation of threads per process on your system?

**Comment by Legodev** [comment]
I doubt so, I also have no settings that prevent fork bombs. The only possibility would be the limited stack size, so I set it to unlimited: A90WO5EI60.
```
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
scheduling priority             (-e) 0
file size               (blocks, -f) unlimited
pending signals                 (-i) 128294
max locked memory       (kbytes, -l) 64
max memory size         (kbytes, -m) unlimited
open files                      (-n) 1024
pipe size            (512 bytes, -p) 8
POSIX message queues     (bytes, -q) 819200
real-time priority              (-r) 0
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 128294
virtual memory          (kbytes, -v) unlimited
file locks                      (-x) unlimited
```

**Comment by vpltd-kgalaj** [comment]
And once again it crashed in thread 64, and once again in libpthread...

Thread 64 (crashed)
0  0x8c6b5164
eip = 0x8c6b5164   esp = 0x8d2de704   ebp = 0x8d2de750   ebx = 0x8d2e9001
esi = 0x8c5321e8   edi = 0x8c53224c   eax = 0x0000271a   ecx = 0x00000000
edx = 0x00000001   efl = 0x00210296
Found by: given as instruction pointer in context
1  0x8c6b52d8
eip = 0x8c6b52d8   esp = 0x8d2de758   ebp = 0x8d2de7dc
Found by: previous frame's frame pointer
2  0x4530de0
eip = 0x04530de0   esp = 0x8d2de7e4   ebp = 0x00000000
Found by: previous frame's frame pointer
3  libpthread-2.22.so + 0x8739
eip = 0x55b70739   esp = 0x8d2de7ec   ebp = 0x8d2de820
Found by: stack scanning

**Comment by Legodev** [comment]
Maybe it is an incompatibility with glibc, I will compile another version.

**Comment by Legodev** [comment]
Well, as I said an good log would help a lot! :D

I created an folder in tmp, extracted an older glibc there, configured and compiled it:
```
../glibc-2.21/configure --prefix /tmp/glibc/ --host=i686-linux-gnu --build=i686-linux-gnu CC="gcc -m32" CXX="g++ -m32" CFLAGS="-O2 -march=i686" CXXFLAGS="-O2 -march=i686"
make 
...
```

Then started the game and it worked. Now i will check if the engine does not work with the glibc-2.22 or if it does not work with an extended glibc (configured with more features then the default one):
```
export LD_LIBRARY_PATH=/tmp/glibc/lib/
/dirt.i386 
Setting breakpad minidump AppID = 201700
dlopen failed trying to load:
/home/lego/.local/share/Steam/linux32/libsteam.so
with error:
/home/lego/.local/share/Steam/linux32/libsteam.so: cannot open shared object file: No such file or directory
dlopen failed trying to load:
libsteam.so
with error:
libsteam.so: cannot open shared object file: No such file or directory
Steam_SetMinidumpSteamID:  Caching Steam ID:  ... [API loaded no]
AL lib: (EE) alc_cleanup: 1 device not closed
```

**Comment by vpltd-peterm** [comment]
As you're using Gentoo, it's quite likely some odd incompatibility between the libs in steam-runtime and your custom glibc. Are you using the steam runtime, or have you disabled it?

**Comment by Legodev** [comment]
Yeah I do use the steam-runtime, I never tried what happens if I disable the runtime use flag .

-> I already had to delete an SDL file, because another game would not launch with the runtime Version. But since this was also an bug in other Distributions I did not recognise this as incompatibility between runtime and my os.
